module/persistent.lua

```diff
@@ -0,0 +1,192 @@
+local awful = require("awful")
+local gears = require("gears")
+
+local capi =
+{
+    awesome = awesome,
+    root = root,
+    screen = screen,
+    client = client
+}
+local ipairs = ipairs
+local table = table
+local tonumber = tonumber
+local tostring = tostring
+
+local persistent = { mt = {} }
+local instance = nil
+
+local function get_xproperty(name, type)
+    capi.awesome.register_xproperty(name, type)
+    return capi.awesome.get_xproperty(name)
+end
+
+local function set_xproperty(name, type, value)
+    capi.awesome.register_xproperty(name, type)
+    capi.awesome.set_xproperty(name, value)
+end
+
+local function client_get_xproperty(client, name, type)
+    local xprop = "bling.client." .. name
+    capi.awesome.register_xproperty(xprop, type)
+    return client:get_xproperty(xprop)
+end
+
+local function client_set_xproperty(client, name, type, value)
+    local xprop = "bling.client." .. name
+    capi.awesome.register_xproperty(xprop, type)
+    client:set_xproperty(xprop, value)
+end
+
+local function save()
+    set_xproperty("tag_index", "number", awful.screen.focused().selected_tag.index)
+
+    for index, entry in ipairs(capi.root.tags()) do
+        local master_width_property = "tag_" .. entry.index .. "_master_width_factor"
+        set_xproperty(master_width_property, "string", tostring(entry.master_width_factor))
+
+        local layout_property = "tag_" .. entry.index .. "_layout"
+        set_xproperty(layout_property, "number", awful.layout.get_tag_layout_index(entry))
+    end
```

Comment:
Can this be extended to support more properties from tags?

I wounder how far we can push this to support dynamic tags.

Reply:
> Can this be extended to support more properties from tags?

I don't see why not. Is there anything I could adad?

> I wounder how far we can push this to support dynamic tags.

I'd suppose it would be quite close to what I'm doing to save the bling tabs?

Reply:
I guess, all of them, :shrug: would it make sense?

My ideal solution would be to create tags from the restore function. (The explanation for that is that I don't have tags by default in my config, I only create them dynamically, so the current implementation wouldn't work for me)